corehq/ex-submodules/casexml/apps/case/tests/util.py

```diff
@@ -29,7 +29,16 @@ def __init__(self, xml_element, version=V2):
         self.version = version
 
     def to_string(self):
-        return ElementTree.tostring(self.xml_element)
+        restore_element_copy = ElementTree.fromstring(ElementTree.tostring(self.xml_element))
+        # hack need to explicitly convert date_opened from datetimes to dates in restores
+        # because the mobile can't handle datetimes.
+        # see: http://manage.dimagi.com/default.asp?230043 for more info
+        date_opened_element = restore_element_copy.find(
+            '{http://commcarehq.org/case/transaction/v2}update/{http://commcarehq.org/case/transaction/v2}date_opened'
+        )
+        if date_opened_element is not None and date_opened_element.text:
+            date_opened_element.text = date_opened_element.text[:10]
```

Comment:
Yeah it is. It turns out that if it's a `date` we actually fail hard on receiving the submission right now (because it parses to a `date` and later tries to save it to a `datetime` property). The other option was to coerce it to a `datetime` during submission and always make it a `date` in tests - which I guess now that you mention it could be better. We would just need to make sure that we add a test for the situation of submitting it as a `datetime` and restoring it as a `date`.

@mkangia we can look at this today.